atom/browser/atom_javascript_dialog_manager.cc

```diff
@@ -41,12 +51,20 @@ void AtomJavaScriptDialogManager::RunJavaScriptDialog(
     buttons.push_back("Cancel");
   }
 
+  origin_counts_[origin]++;
+
+  std::string checkbox_string = "";
+  if (origin_counts_[origin] > 1 &&
+      WebContentsPreferences::UsesSafeDialogs(web_contents)) {
+    checkbox_string = "Prevent this app from creating additional dialogs";
```

Comment:
Hmmm how do we handle i18n for this?

I'm not aware of any other user-facing strings like this in Atom. There are thrown errors, but those are intended more for developers than end-users and so don't have translations.

Reply:
I saw some other cases like int the `web_dialog_helper.cc` for example. chrome/ui has `l10n` utlis namespace which provides localization support, but that might be a whole different thing.

Reply:
Maybe we can make the string user defined in `webPreferences`?  Then it's up to implementers to both choose the string and translate it?

Reply:
Since this string has already been translated in Chromium, we can build the `components_strings` target in libcc and then ship the `components_strings_$locale.pak` files, which can be then loaded in Electron to provide localized strings.

And I think it is fine not to do localization inside Electron for now, we already have some English-only strings (for example "All Files") and this does not make things worse.